src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -1481,13 +1490,15 @@ namespace
 
 void MainWindow::slot_clear_render_region()
 {
-    m_rendering_manager.schedule(
-        auto_ptr<RenderingManager::IScheduledAction>(
-            new ClearRenderRegionAction()));
+    auto_ptr<RenderingManager::IScheduledAction> clear_render_region_action(
+        new ClearRenderRegionAction(this));
 
-    m_rendering_manager.reinitialize_rendering();
+    if (m_rendering_manager.is_rendering())
```

Comment:
Actually, we have two possible cases right here. 
First, the button is clicked while rendering is on the go. Then, `ClearRenderRegionAction` should be spawned and put to the `RenderingManager` task queue. This is the one that was implemented. 
However, if we are not rendering, it won't take the effect until the rendering process will be started. Also, if the button will be clicked like thousand times, you'll get thousand tasks in queue at the beginning of rendering process.
So, the second case is to properly handle button click when rendering process is stopped.